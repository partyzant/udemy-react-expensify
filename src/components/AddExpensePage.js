import React from 'react';

const AddExpensePage = () => (
  <div>
    this is from my AddExpensePage
  </div>
);

export default AddExpensePage;
