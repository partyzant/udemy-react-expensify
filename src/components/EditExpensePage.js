import React from 'react';

const EditExpensePage = (props) => {
  console.log(props);
  return (
    <div>
      this is from my EditExpensePage {props.match.params.id}
    </div>
  )
};

export default EditExpensePage;
