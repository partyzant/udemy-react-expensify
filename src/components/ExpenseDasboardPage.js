import React from 'react';

const ExpenseDasboardPage = () => (
  <div>
    this is from my ExpenseDasboardPage
  </div>
);

export default ExpenseDasboardPage;
