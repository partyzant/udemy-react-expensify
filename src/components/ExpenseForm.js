import React from 'react';
import moment from 'moment';
import 'react-dates/initialize';
import {SingleDatePicker} from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';

moment.locale('pl');
const now = moment();
console.log(now.format('MMM Do, YYYY'))

export default class ExpenseForm extends React.Component {
  state = {
    description: '',
    note: '',
    amount: '',
    createdAt: moment(),
    calendarFocused: false,
    error: ''
  };
  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState(() => ({description}));
  };
  onNoteChange = (e) => {
    // const note = e.target.value;
    e.persist(); // alternatywny sposób przekazywania wartości
    this.setState(() => ({note: e.target.value}));
  };
  onAmountChange = (e) => {
    const amount = e.target.value;

    if(!amount || amount.match(/^\d+(\.\d{0,2})?$/)) {
      this.setState(() => ({amount}));
    }
  };
  onDateChange = (createdAt) => {
    if(createdAt) {
      this.setState({ createdAt })
    }
  };
  onSubmit = (e) => {
    e.preventDefault();

    if(!this.state.description || !this.state.amount){
      this.setState({error: 'Wpisz description i kwotę!'});
    } else {
      this.setState({error: ''});
      this.props.onSubmit({
        description: this.state.description,
        amount: parseFloat(this.state.amount, 10) * 100,
        createdAt: this.state.createdAt.valueOf(),
        note: this.state.note
      });
    }
  };
  render () {
    return (
      <div>
        {this.state.error && <p style={{color:'red'}}>{this.state.error}</p>}
        <form onSubmit={this.onSubmit}>
          <input
            type="text"
            placeholder="Description"
            autoFocus
            value={this.state.description}
            onChange={this.onDescriptionChange}
          /><br/>
          <input
            type="text"
            placeholder="Amount"
            value={this.state.amount}
            onChange={this.onAmountChange}
          /><br/>
          <SingleDatePicker
            date={this.state.createdAt} // momentPropTypes.momentObj or null
            onDateChange={this.onDateChange} // PropTypes.func.isRequired
            focused={this.state.calendarFocused} // PropTypes.bool
            onFocusChange={({ focused }) => this.setState({ calendarFocused: focused })} // PropTypes.func.isRequired
            numberOfMonths={1}
            isOutsideRange={() => false}
          /><br/>
          <textarea
            placeholder="Add note (optional)"
            value={this.state.note}
            onChange={this.onNoteChange}
          >
          </textarea><br/>
          <button>Add Expense</button>
        </form>
      </div>
    );
  }
};
