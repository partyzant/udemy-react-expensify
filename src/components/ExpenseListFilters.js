import React from 'react';
import {connect} from 'react-redux';
import {setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate} from '../actions/filters';
import moment from 'moment';
import 'react-dates/initialize';
import {DateRangePicker} from 'react-dates';
// import 'react-dates/lib/css/_datepicker.css';

class ExpenseListFilters extends React.Component {
  state = {
    focusedInput: null
  };
  onDatesChange = ({ startDate, endDate }) => {
    // console.log(setStartDate(startDate))
    // console.log(setEndDate(endDate))
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  };
  render() {
    return(
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          onChange={(e) => {
            this.props.dispatch(setTextFilter(e.target.value));
          }}
        />
        <select
          value={this.props.filters.sortBy}
          onChange={(e) => {
            this.props.dispatch( e.target.value === 'date' ? sortByDate() : sortByAmount() );
          }}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <br/>
        <DateRangePicker
          startDate={this.props.filters.startDate} // momentPropTypes.momentObj or null,
          endDate={this.props.filters.endDate} // momentPropTypes.momentObj or null,
          onDatesChange={this.onDatesChange} // PropTypes.func.isRequired,
          focusedInput={this.state.focusedInput} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
          onFocusChange={focusedInput => this.setState({ focusedInput })} // PropTypes.func.isRequired,
          isOutsideRange={() => false}
          numberOfMonths={1}
          showClearDates={true}
        />
      </div>
    );
  }
}
const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

export default connect(mapStateToProps)(ExpenseListFilters);
