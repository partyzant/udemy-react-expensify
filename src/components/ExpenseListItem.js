import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

export default ( { id, description, amount, createdAt } ) => {
  return(
  <div>
      <h3>
        <Link to={`/edit/${id}`}>{description}</Link>
      </h3>

    <p>{amount} - {moment(createdAt).format('YYYY MM DD')}</p>
  </div>
);
}
// export default ExpenseListItem;
