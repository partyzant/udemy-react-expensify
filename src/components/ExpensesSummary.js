import React from 'react';
import numeral from 'numeral';

export const ExpensesSummary = ({expenseCount, expensesTotal}) => {
  const formatedExpTot = numeral(expensesTotal / 100).format('$0,0.00');
  return (
    <div>
      <h1>Viewing {expenseCount} totaling {formatedExpTot}</h1>
    </div>
  );
};
