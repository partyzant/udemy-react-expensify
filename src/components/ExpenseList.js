import React from 'react';
import {connect} from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import { ExpensesSummary } from './ExpensesSummary';
import selectExpenses from '../selectors/expenses'
import { log } from 'util';
import selectExpensesTotal from '../selectors/expenses-total'


export const ExpenseList = (props) => {
  const expenseCount = props.expenses.length;
  const expensesTotal = selectExpensesTotal(props.expenses);

  return (
    <div>
      <div>
        <ExpensesSummary expenseCount={expenseCount} expensesTotal={expensesTotal}/>
      </div>
      {
        props.expenses.length === 0 ? (
          <p>No expenses</p>
        ) : (
          props.expenses.map((expense, i) => <ExpenseListItem key={i} {...expense} />)
        )
      }
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);
