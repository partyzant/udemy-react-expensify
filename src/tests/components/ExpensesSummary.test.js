import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('should render expenses total with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={14560} />);
    expect(wrapper).toMatchSnapshot();  
});