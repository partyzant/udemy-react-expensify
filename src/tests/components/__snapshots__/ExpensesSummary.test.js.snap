// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenses total with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     totaling 
    $145.60
  </h1>
</div>
`;
